.github/workflows/comment-cfg-grammar-changes.yml

```diff
@@ -0,0 +1,92 @@
+name: Comment config grammar changes (PR)
+
+on:
+  workflow_run:
+    workflows: [Check config grammar changes (PR)]
+    types:
+      - completed
+
+permissions:
+  pull-requests: write
+
+jobs:
+  update-or-remove-comment-on-pr:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+        # The download-artifact action cannot download artifacts from other workflows.
+        # Copied from https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run
+      - name: Download comment artifact
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "comment"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/comment.zip`, Buffer.from(download.data));
+
+      - name: Unzip comment artifact
+        run: unzip comment.zip
+
+      - name: Get PR ID
+        id: pr-id
+        run: |
+          # The pull_requests array always come empty when querying a workflow run's data if it is started from a fork. This might be a GitHub bug.
+          # There is also no json field in the `gh run view` command, which could give us the PR ID, so we can only hack it from the CLI output...
+          # See https://github.com/orgs/community/discussions/25220
+
+          PR_ID=$(gh run view -R ${{ github.repository }} ${{ github.event.workflow_run.id }} | grep -oP '#[0-9]+ . ${{ github.event.workflow_run.id }}' | grep -oP '#[0-9]+' | cut -c 2-)
```

Comment:
@OverOrion suggested a better way to workaround this problem, I will do that soon. The PR is ready to review otherwise.

Reply:
Done.